git_machete/github.py

```diff
@@ -24,9 +25,35 @@ def __repr__(self) -> str:
 GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
 
 
-def github_token() -> Optional[str]:
+def _token_from_gh() -> Optional[str]:
+    proc = subprocess.run(
+        ["gh", "auth", "status", "--show-token"],
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+
+    # gh auth status emits outputs to stdout
+    result = proc.stderr.decode()
+
+    match = re.search("Token: (\w+)", result)
```

Comment:
Too bad the output is so brittle... I hope it doesn't change between versions? 🤔 

For 1.11.0, it's:

```
github.com
  ✓ Logged in to github.com as PawelLipski (GITHUB_TOKEN)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: .....
```

Reply:
I've noticed you had lowercase `token` previously, right? was it different in some older version?

Reply:
Yeah, agreed. We're probably "in the porcelain" on this one, but it's a *very* convenient way of managing and retrieving auth tokens.

I think by restricting outselves to the `Token: (\w+)` regex we'll be less sensitive. 
So, while it may break in the future, I think the risk is reasonably low and the impact is just a "observable loss of function".